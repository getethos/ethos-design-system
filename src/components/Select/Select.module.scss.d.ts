// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'Compact': string;
  'root': string;
  'title': string;
  'wrapper': string;
}
export const cssExports: CssExports;
export default cssExports;
