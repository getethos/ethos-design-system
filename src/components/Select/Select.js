import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types'
// See React-Select -- https://github.com/JedWatson/react-select for documentation
// on usage, Async configuration, etc.
import ReactSelect, { components } from 'react-select'
import ReactSelectAsync from 'react-select/async'
import ReactSelectAsyncCreatable from 'react-select/async-creatable'
import ReactSelectCreatable from 'react-select/creatable'
import useErrorMessage from '../../hooks/useErrorMessage.js'
import { InputLabel } from '../InputLabel'

import styles from './Select.module.scss'

export const ReactSelectComponents = components

/**
 * Component which wraps [react-select](https://github.com/JedWatson/react-select). Note,
 * within the form engine, your validator should be able to handle the two potential shapes
 * react-select calls [onChange with](https://react-select.com/props#statemanager-props).
 * TL;DR is single select gets an object like:
 * {value: "abc", label: "abc"}, and multi: [{value: "abc", label: "abc"}]
 * So, a multi-select validator would likely want to loop and validate each items one by one.
 */
export const Select = ({
  className,
  title,
  isAsync,
  isCompact,
  isCreatable,
  validator,
  onChange,
  formChangeHandler,
  currentError,
  formTouched,
  labelCopy,
  allCaps = true,
  name,
  ...rest
}) => {
  const resolvedValidator = validator ? validator : () => ''
  const [getError, setError, , validate] = useErrorMessage(resolvedValidator)
  const compactClass = isCompact ? styles.Compact : ''

  // According to the react-select https://react-select.com/props#statemanager-props docs
  // type will be: one of<Object, Array<Object>, null, undefined>,
  // For single (object) in shape of:
  //    {value: "CA", label: "CA"}
  // for multi (array):
  //    [{"value": "CA", "label": "CA"}, {"value": "NY", "label": "NY"}]
  const [userSelection, updateUserSelection] = useState(undefined)
  const onChangeHandler = (lastSelection, actionMeta) => {
    /**
     * For multi selects, react-select allows the user to remove all the
     * selected items, and once there are no more, this will be `null`. But,
     * for our state and validation purposes, it's more convenient to resolve
     * this to an empty array here. This will in turn result in the form engine's
     * validator getting called with an empty array which is easier for it to do
     * meaningful validation against since it's already expecting an array.
     */
    if (!lastSelection) {
      // This condition should not happen for single selects because once user has
      // selected a value, react-select prevent removing. So only form multi selects.
      updateUserSelection([])
    } else {
      updateUserSelection(lastSelection)
    }
    if (onChange) {
      onChange(lastSelection, actionMeta)
    }
  }

  const props = {
    className: `${className ? className : ''} ${compactClass} ${styles.root}`,
    onChange: onChangeHandler,
    onBlur,
    'aria-label': title, // https://react-select.com/props#select-props
    ...rest,
  }

  const validationSelect = () => {
    let errorMessage = ''
    let resolvedValues = ''
    // react-select multi select case
    if (Array.isArray(userSelection)) {
      resolvedValues = userSelection.map((selection) => selection.value)
    } else {
      // react-select single select case
      resolvedValues = userSelection.value
    }

    errorMessage = validate(resolvedValues)
    setError(errorMessage)
    if (formChangeHandler) {
      formChangeHandler(resolvedValues, errorMessage)
    }
  }

  useEffect(() => {
    const isSelectedValue = typeof userSelection !== 'undefined'
    if (isSelectedValue) {
      validationSelect()
    }
  }, [userSelection])

  const onBlur = () => {
    validationSelect()
  }

  const wrapperClass = title ? styles.wrapper : ''

  const getTag = () => {
    if (isAsync && isCreatable) {
      return ReactSelectAsyncCreatable
    } else if (isAsync) {
      return ReactSelectAsync
    } else if (isCreatable) {
      return ReactSelectCreatable
    } else {
      return ReactSelect
    }
  }

  const SelectTag = getTag()

  return (
    <div className={wrapperClass} data-tid={rest['data-tid']}>
      {labelCopy && (
        <InputLabel name={name} labelCopy={labelCopy} allCaps={allCaps} />
      )}
      <SelectTag {...props} />
      {title && <div className={styles.title}>{title}</div>}
      {getError(currentError, formTouched)}
    </div>
  )
}

Select.propTypes = {
  /** class name prefix */
  classNamePrefix: PropTypes.string,
  /** loadOptions should take an inputValue and return a Promise that resolves */
  // to an array of options.
  loadOptions: PropTypes.func,

  /** The last user's selection which will be one of:
   *  `<Object, Array<Object>, null, undefined>` according to the react-select docs.
   *  For single (object) in shape of: {value: "CA", label: "CA"}
   *  For multi (array): [{"value": "CA", "label": "CA"}, ...]
   */
  onChange: PropTypes.func,
  /** Is asynchronous (see react-select docs) */
  isAsync: PropTypes.bool,
  /** Is compact or shorter */
  isCompact: PropTypes.bool,
  /** The title for this select */
  title: PropTypes.string,
  /** className */
  className: PropTypes.string,
  /** Is creatable (see react-select docs) */
  isCreatable: PropTypes.bool,
  formChangeHandler: PropTypes.func,
  currentError: PropTypes.string,
  formTouched: PropTypes.bool,
  labelCopy: PropTypes.string,
  allCaps: PropTypes.bool,
  name: PropTypes.string,
  validator: PropTypes.func,
}

Select.defaultProps = {
  classNamePrefix: 'StyledReactSelect', // for styling
  className: undefined,
  placeholder: 'Type to search',
}
