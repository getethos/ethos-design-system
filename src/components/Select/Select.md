We leverage [React-Select](https://github.com/JedWatson/react-select) for our single and multi selects. For example, you can add
`menuIsOpen={true}` below as an attribute to `<Select>` and the menu will remain open.
```jsx
const onSelected = selectedOption => {
  console.log('Option selected: ', selectedOption)
};
const options = [
  { value: 'nyc', label: 'New York' },
  { value: 'sf', label: 'San Francisco' },
  { value: 'chicago', label: 'Chicago' },
  { value: 'phoenix', label: 'Phoenix' },
  { value: 'la', label: 'Los Angeles' },
  { value: 'seattle', label: 'Seattle' },
  { value: 'Portland', label: 'Portland' },
  { value: 'Miami', label: 'Miami' },
];
<Select
  placeholder='Custom placeholder...'
  onChange={onSelected}
  options={options}
/>
```

An optional title may be included on the input box
```jsx
const onSelected = selectedOption => {
  console.log('Option selected: ', selectedOption)
};
const options = [
  { value: 'nyc', label: 'New York' },
  { value: 'sf', label: 'San Francisco' },
  { value: 'chicago', label: 'Chicago' },
  { value: 'phoenix', label: 'Phoenix' },
  { value: 'la', label: 'Los Angeles' },
  { value: 'seattle', label: 'Seattle' },
  { value: 'Portland', label: 'Portland' },
  { value: 'Miami', label: 'Miami' },
];
<Select
  placeholder='Custom placeholder...'
  onChange={onSelected}
  options={options}
  title={'Input Title (optional)'}
/>
```

```jsx
import { COLORS } from '../Colors'
const colourOptions = [{
  value: 'brandforest',
  label: 'Brand Forest',
  color: COLORS.BRAND_FOREST,
  isFixed: true
}, {
  value: 'salamander',
  label: 'Brand Salamander (Disabled)',
  color: COLORS.BRAND_SALAMANDER,
  isDisabled: true
}, {
  value: 'grayprimary',
  label: 'Gray Primary',
  color: COLORS.GRAY_PRIMARY,
}, {
  value: 'graystrokedisabled',
  label: 'Gray Stroke and Disabled',
  color: COLORS.GRAY_STROKE_AND_DISABLED,
}, {
  value: 'pink',
  label: 'Pink',
  color: 'pink'
}, {
  value: 'green',
  label: 'Green',
  color: 'green'
}, {
  value: 'red',
  label: 'Red',
  color: 'red'
}, {
  value: 'blue',
  label: 'Blue',
  color: 'blue',
}];

<Select
  defaultValue={[colourOptions[3]]}
  isMulti
  name="colors"
  options={colourOptions}
  menuIsOpen={true}
/>
```
