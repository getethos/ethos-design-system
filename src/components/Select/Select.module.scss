.root {
  --remove: #d6d6d6;
  --remove-hover: #c6c6c6;

  :global {
    .StyledReactSelect__value-container {
      font-family: var(--Theinhardt-font-stack);
      font-size: calc(var(--base-font-size) * 1px);
      line-height: calc(var(--base-line-height) / var(--base-font-size));
      overflow: visible;
      padding: var(--Space-8) var(--Space-16);
      /* Force to be consistent with TextInput */
      min-height: var(--Space-56);
    }

    .StyledReactSelect__multi-value {
      margin: 0;
      padding: 2px;
      margin-left: 0;
      margin-bottom: 0;
      margin-top: 2px;
      margin-right: var(--Space-6);
    }

    .StyledReactSelect__value-container > div:last-child {
      /* This is an anonymous wrapper around __input. */
      margin-top: 0;
      margin-bottom: 0;
      padding-top: 0;
      padding-bottom: 0;
    }

    .StyledReactSelect__input {
      margin-left: 0;

      input[style] {
        font-size: calc(var(--base-font-size) * 1px);
        line-height: calc(var(--base-line-height) / var(--base-font-size));
        padding: 0;
        width: 4em !important; /* overrides lib inline style *sigh* */
      }
    }

    .StyledReactSelect__placeholder {
      font-size: calc(var(--base-font-size) * 1px);
      line-height: calc(var(--base-line-height) / var(--base-font-size));
      margin: 0;
    }

    .StyledReactSelect__option--is-focused {
      cursor: pointer;
    }
    .StyledReactSelect__option--is-focused,
    .StyledReactSelect__option:hover {
      background: var(--BrandMoss);
    }

    /* Do not hover highlight if disabled option */
    .StyledReactSelect__option:hover.StyledReactSelect__option--is-disabled {
      background: transparent;
    }

    .StyledReactSelect__option--is-selected,
    .StyledReactSelect__option--is-selected:hover {
      background: var(--BrandForest);
    }

    .StyledReactSelect__control {
      border-radius: 0;
      border-style: solid;
      border-width: 1px;
      box-shadow: none;
      outline: 0;
      transition: border-color 250ms ease;

      &,
      &:hover {
        border-color: var(--GrayStrokeAndDisabled--translucent);
      }

      &--is-focused {
        &,
        &:hover {
          border-color: var(--GrayStrokeAndDisabled--translucent);
        }
      }
    }

    .StyledReactSelect__multi-value__remove {
      background-color: var(--remove);

      &:hover {
        background-color: var(--remove-hover);
        cursor: pointer;
        /* overrides library's ugly red 'X' */
        color: inherit;
      }
    }

    .StyledReactSelect__input {
      input {
        color: var(--GrayPrimary--opaque);
      }
    }

    .StyledReactSelect__placeholder {
      color: var(--GrayStrokeAndDisabled--opaque);
    }
  }
}

.title {
  position: relative;
  height: 0;
  top: -48px;
  left: 18px;
  font-size: 12px;
  color: var(--GraySecondary--translucent);
  pointer-events: none;
}

.wrapper {
  :global {
    .StyledReactSelect__value-container {
      top: 9px;
    }
  }
}

.Compact {
  :global {
    .StyledReactSelect__input,
    .StyledReactSelect__control,
    .StyledReactSelect__placeholder,
    .StyledReactSelect__option,
    .StyledReactSelect__value-container {
      font-size: 15px;
    }

    .StyledReactSelect__control {
      box-sizing: border-box;
      min-height: var(--Space-40);
      border-radius: 3px;
    }
    .StyledReactSelect__value-container {
      font-size: 15px;
      /* Force to be consistent with TextInput */
      min-height: var(--Space-34);
      /* Since we've made it compact we need to re-center */
      padding: var(--Space-6) var(--Space-16) var(--Space-8) var(--Space-16);
    }
  }
}
