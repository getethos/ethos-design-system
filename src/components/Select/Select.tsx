import * as React from 'react'
import { useEffect, useState } from 'react'
// See React-Select -- https://github.com/JedWatson/react-select for documentation
// on usage, Async configuration, etc.
import * as ReactSelect from 'react-select'
import { components, Props as ReactSelectProps } from 'react-select'
import * as ReactSelectAsync from 'react-select/async'
import * as ReactSelectAsyncCreatable from 'react-select/async-creatable'
import * as ReactSelectCreatable from 'react-select/creatable'
import useErrorMessage from '../../hooks/useErrorMessage'
import { PublicFormProps } from '../Form/Form'
import { InputLabel } from '../InputLabel'
import styles from './Select.module.scss'

type UserSelection = { label: string; value: string }

export const ReactSelectComponents: {
  ClearIndicator: any
  Control: any
  DropdownIndicator: any
  DownChevron: any
  CrossIcon: any
  Group: any
  GroupHeading: any
  IndicatorsContainer: any
  IndicatorSeparator: any
  Input: any
  LoadingIndicator: any | null
  Menu: any
  MenuList: any
  MenuPortal: any
  LoadingMessage: any
  NoOptionsMessage: any
  MultiValue: any
  MultiValueContainer: any
  MultiValueLabel: any
  MultiValueRemove: any
  Option: any
  Placeholder: any
  SelectContainer: any
  SingleValue: any
  ValueContainer: any
} = components

type SelectProps = PublicFormProps &
  ReactSelectProps & {
    classNamePrefix?: string
    isAsync?: boolean
    isCompact?: boolean
    title?: string
    className?: string
    isCreatable?: boolean
    labelCopy?: string
    allCaps?: boolean
  }
/**
 * Component which wraps [react-select](https://github.com/JedWatson/react-select). Note,
 * within the form engine, your validator should be able to handle the two potential shapes
 * react-select calls [onChange with](https://react-select.com/props#statemanager-props).
 * TL;DR is single select gets an object like:
 * {value: "abc", label: "abc"}, and multi: [{value: "abc", label: "abc"}]
 * So, a multi-select validator would likely want to loop and validate each items one by one.
 */
export const Select: React.SFC<SelectProps> = ({
  className,
  title,
  isAsync,
  isCompact,
  isCreatable,
  validator,
  onChange,
  formChangeHandler,
  currentError,
  formTouched,
  labelCopy,
  allCaps = true,
  name,
  ...rest
}) => {
  const resolvedValidator = validator ? validator : () => ''
  const [getError, setError, , validate] = useErrorMessage(resolvedValidator)
  const compactClass = isCompact ? styles.Compact : ''
  // According to the react-select https://react-select.com/props#statemanager-props docs
  // type will be: one of<Object, Array<Object>, null, undefined>,
  // For single (object) in shape of:
  //    {value: "CA", label: "CA"}
  // for multi (array):
  //    [{"value": "CA", "label": "CA"}, {"value": "NY", "label": "NY"}]
  const [userSelection, updateUserSelection] = useState<
    UserSelection | UserSelection[] | undefined
  >(undefined)
  const onChangeHandler = (lastSelection, actionMeta) => {
    /**
     * For multi selects, react-select allows the user to remove all the
     * selected items, and once there are no more, this will be `null`. But,
     * for our state and validation purposes, it's more convenient to resolve
     * this to an empty array here. This will in turn result in the form engine's
     * validator getting called with an empty array which is easier for it to do
     * meaningful validation against since it's already expecting an array.
     */
    if (!lastSelection) {
      // This condition should not happen for single selects because once user has
      // selected a value, react-select prevent removing. So only form multi selects.
      updateUserSelection([])
    } else {
      updateUserSelection(lastSelection)
    }
    if (onChange) {
      onChange(lastSelection, actionMeta)
    }
  }
  const onBlur = () => {
    validationSelect()
  }
  const props = {
    className: `${className ? className : ''} ${compactClass} ${styles.root}`,
    onChange: onChangeHandler,
    onBlur,
    'aria-label': title,
    ...rest,
  }
  const validationSelect = () => {
    if (!userSelection) return
    let errorMessage = ''
    let resolvedValues: string | string[] = ''
    // react-select multi select case
    if (Array.isArray(userSelection)) {
      resolvedValues = userSelection.map((selection) => selection.value)
    } else {
      // react-select single select case
      resolvedValues = userSelection.value
    }
    errorMessage = validate(resolvedValues)
    setError(errorMessage)
    if (formChangeHandler) {
      formChangeHandler(resolvedValues, errorMessage)
    }
  }
  useEffect(() => {
    const isSelectedValue = typeof userSelection !== 'undefined'
    if (isSelectedValue) {
      validationSelect()
    }
  }, [userSelection])
  const wrapperClass = title ? styles.wrapper : ''
  const getTag = () => {
    if (isAsync && isCreatable) {
      return ReactSelectAsyncCreatable
    } else if (isAsync) {
      return ReactSelectAsync
    } else if (isCreatable) {
      return ReactSelectCreatable
    } else {
      return ReactSelect
    }
  }
  const SelectTag = (getTag() as unknown) as React.ElementType
  return (
    <div className={wrapperClass} data-tid={rest['data-tid']}>
      {labelCopy && (
        <InputLabel name={name} labelCopy={labelCopy} allCaps={allCaps} />
      )}
      <SelectTag {...props} />
      {title && <div className={styles.title}>{title}</div>}
      {getError(currentError, formTouched)}
    </div>
  )
}
Select.defaultProps = {
  classNamePrefix: 'StyledReactSelect',
  className: undefined,
  placeholder: 'Type to search',
}
