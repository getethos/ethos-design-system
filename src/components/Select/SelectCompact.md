We leverage [React-Select](https://github.com/JedWatson/react-select) for our single and multi selects. For example, you can add
`menuIsOpen={true}` below as an attribute to `<Select>` and the menu will remain open.

```jsx
const onSelected = (selectedOption) => {
  console.log('Option selected: ', selectedOption)
}
const options = [
  { value: 'nyc', label: 'New York' },
  { value: 'sf', label: 'San Francisco' },
  { value: 'chicago', label: 'Chicago' },
  { value: 'phoenix', label: 'Phoenix' },
  { value: 'la', label: 'Los Angeles' },
  { value: 'seattle', label: 'Seattle' },
  { value: 'Portland', label: 'Portland' },
  { value: 'Miami', label: 'Miami' },
]
;<SelectCompact
  placeholder="Custom placeholder..."
  onChange={onSelected}
  options={options}
  className="StyledReactSelect-Compact"
/>
```
