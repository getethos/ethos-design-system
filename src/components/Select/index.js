export { Select } from './Select.js'
