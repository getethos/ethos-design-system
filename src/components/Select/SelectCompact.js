import React from 'react'
import PropTypes from 'prop-types'
import { Select } from './Select'
import styles from './SelectCompact.module.scss'

export const SelectCompact = (props) => {
  return <Select {...props} />
}

Select.propTypes = {
  className: PropTypes.string,
}

Select.defaultProps = {
  className: 'StyledReactSelect-Compact',
}
