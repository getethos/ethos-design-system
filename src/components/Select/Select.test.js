import React from 'react'
import { Select } from './Select.js'
import { render } from '@testing-library/react'

describe('Select', () => {
  const mockProps = {
    className: 'foo',
    isCompact: true,
    'data-tid': 'my-select',
  }

  test.only('matches snapshot', () => {
    const tree = render(<Select {...mockProps} />)
    expect(tree).toMatchSnapshot()
  })

  test('adding a title prop adds a div and aria-label', () => {
    const tree = renderer
      .create(<Select {...mockProps} title="Hello World" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
