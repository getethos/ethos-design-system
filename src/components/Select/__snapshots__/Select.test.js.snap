// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select adding a title prop adds a div and aria-label 1`] = `
<div
  className="wrapper"
  data-tid="my-select"
>
  <div
    className="foo Compact root css-2b097c-container"
    onKeyDown={[Function]}
  >
    <div
      className="StyledReactSelect__control css-yk16xz-control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="StyledReactSelect__value-container css-1hwfws3"
      >
        <div
          className="StyledReactSelect__placeholder css-1wa3eu0-placeholder"
        >
          Type to search
        </div>
        <div
          className="css-b8ldur-Input"
        >
          <div
            className="StyledReactSelect__input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              aria-label="Hello World"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="react-select-3-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "label": "input",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="StyledReactSelect__indicators css-1hb7zxy-IndicatorsContainer"
      >
        <span
          className="StyledReactSelect__indicator-separator css-1okebmr-indicatorSeparator"
        />
        <div
          aria-hidden="true"
          className="StyledReactSelect__indicator StyledReactSelect__dropdown-indicator css-tlfecz-indicatorContainer"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-6q0nyr-Svg"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    className="title"
  >
    Hello World
  </div>
  
</div>
`;

exports[`Select matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=""
        data-tid="my-select"
      >
        <div
          class="foo Compact root css-2b097c-container"
        >
          <div
            class="StyledReactSelect__control css-yk16xz-control"
          >
            <div
              class="StyledReactSelect__value-container css-1hwfws3"
            >
              <div
                class="StyledReactSelect__placeholder css-1wa3eu0-placeholder"
              >
                Type to search
              </div>
              <div
                class="css-b8ldur-Input"
              >
                <div
                  class="StyledReactSelect__input"
                  style="display: inline-block;"
                >
                  <input
                    aria-autocomplete="list"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    id="react-select-2-input"
                    spellcheck="false"
                    style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                  />
                </div>
              </div>
            </div>
            <div
              class="StyledReactSelect__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="StyledReactSelect__indicator-separator css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="StyledReactSelect__indicator StyledReactSelect__dropdown-indicator css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-6q0nyr-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=""
      data-tid="my-select"
    >
      <div
        class="foo Compact root css-2b097c-container"
      >
        <div
          class="StyledReactSelect__control css-yk16xz-control"
        >
          <div
            class="StyledReactSelect__value-container css-1hwfws3"
          >
            <div
              class="StyledReactSelect__placeholder css-1wa3eu0-placeholder"
            >
              Type to search
            </div>
            <div
              class="css-b8ldur-Input"
            >
              <div
                class="StyledReactSelect__input"
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="react-select-2-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class="StyledReactSelect__indicators css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class="StyledReactSelect__indicator-separator css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class="StyledReactSelect__indicator StyledReactSelect__dropdown-indicator css-tlfecz-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-6q0nyr-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
